py/server/deephaven/column.py

```diff
@@ -202,15 +206,30 @@ def datetime_col(name: str, data: Sequence) -> InputColumn:
     Returns:
         a new input column
     """
-    data = [time.to_j_instant(d) for d in data]
+
+    # try to convert to numpy array of datetime64 if not already, so that we can call translateArrayLongToInstant on
+    # it to reduce the number of round trips to the JVM
+    if not isinstance(data, np.ndarray):
+        try:
+            data = np.array([pd.Timestamp(dt).to_numpy() for dt in data], dtype=np.datetime64)
+        except Exception as e:
+            ...
+
+    if isinstance(data, np.ndarray) and data.dtype.kind == 'M':
+        longs = jpy.array('long', data.astype('datetime64[ns]').astype('int64'))
+        data = _JPrimitiveArrayConversionUtility.translateArrayLongToInstant(longs)
```

Comment:
I don't think so. It is more confusing than helpful IMO.

Reply:
I don't follow this logic.  If `data.dtype.kind == 'i'`, you could have an additional block that looks like:
```
if isinstance(data, np.ndarray) and data.dtype.kind == 'i':
        longs = jpy.array('long', data.astype('int64'))
        data = _JPrimitiveArrayConversionUtility.translateArrayLongToInstant(longs)
```
This would just yield one boundary crossing for an array of ints vs what is happening now with a boundary crossing pervalue.

Similarly, there could be another block like this that coerces to a long array instead of a datetime array ... unless the timestamp coercion is already handling this case.  Either way, we need a unit test for epoch nanos ints to make sure they work. 
```
    if not isinstance(data, np.ndarray):
        try:
            data = np.array([pd.Timestamp(dt).to_numpy() for dt in data], dtype=np.datetime64)
        except Exception as e:
            ...
```